File: src/test/testLog4j.java
import org.apache.log4j.Logger;
import org.apache.log4j.PropertyConfigurator;

public class testLog4j {

    public static void main(String[] args) {

//        PropertyConfigurator.configure(testLog4j.class.getResourceAsStream("log4j.properties"));
        PropertyConfigurator.configure(testLog4j.class.getResource("log4j.properties").getPath());

//        testLog4j.class.getResource("log4j.properties").getPath()
//      System.out.println(testLog4j.class.getResource("log4j.properties").getPath());
        Logger log= Logger.getLogger(testLog4j.class);
        log.info("test for log4j");
    }
}

File: src/test/TestLog4j.scala
import org.apache.log4j.{Logger, PropertyConfigurator}

object TestLog4j {
  def main(args: Array[String]): Unit = {

    //PropertyConfigurator.configure(testLog4j.class.getResourceAsStream("config.properties"));
   // PropertyConfigurator.configure(testLog4j.class.getResource("config.properties").getPath());



    PropertyConfigurator.configure(this.getClass.getResource("log4j.properties").getPath)
   // PropertyConfigurator.configure(TestLog4j.getClass.getResource("config.properties").getPath)
   // PropertyConfigurator.configure(TestLog4j.getClass.getResourceAsStream("config.properties"))


    val logger= Logger.getLogger(TestLog4j.getClass)
    logger.info("Hello")
  }
}
